sys/include/sema.h

```diff
@@ -176,8 +191,58 @@ static inline int sema_wait(sema_t *sema)
  */
 static inline int sema_try_wait(sema_t *sema)
 {
-    return _sema_wait(sema, 0, 0);
+#if IS_USED(MODULE_ZTIMER)
+    return _sema_wait_ztimer(sema, 0, NULL, 0);
+#else
+    return _sema_wait_xtimer(sema, 0, 0);
+#endif
+}
+
+#if IS_USED(MODULE_XTIMER) || defined(DOXYGEN)
+/**
+ * @brief   Wait for a semaphore being posted.
+ *
+ * @pre `(sema != NULL)`
+ *
+ * @param[in]  sema     A semaphore.
+ * @param[in]  timeout  Time in microseconds until the semaphore times out.
+ *                      0 does not wait.
+ *
+ * @return  0 on success
+ * @return  -ETIMEDOUT, if the semaphore times out.
+ * @return  -ECANCELED, if the semaphore was destroyed.
+ * @return  -EAGAIN,    if the semaphore is not posted (only if timeout = 0)
+ */
+static inline int sema_wait_timed(sema_t *sema, uint64_t timeout)
+{
+    return _sema_wait_xtimer(sema, (timeout != 0), timeout);
+}
+#endif
```

Comment:
```suggestion
#if IS_USED(MODULE_XTIMER) || IS_USED(MODULE_ZITMER) || defined(DOXYGEN)
/**
 * @brief   Wait for a semaphore being posted.
 *
 * @pre `(sema != NULL)`
 *
 * @param[in]  sema     A semaphore.
 * @param[in]  timeout  Time in microseconds until the semaphore times out.
 *                      0 does not wait.
 *
 * @return  0 on success
 * @return  -ETIMEDOUT, if the semaphore times out.
 * @return  -ECANCELED, if the semaphore was destroyed.
 * @return  -EAGAIN,    if the semaphore is not posted (only if timeout = 0)
 */
static inline int sema_wait_timed(sema_t *sema, uint64_t timeout)
{
#if IS_USED(MODULE_ZTIMER)
    return _sema_wait_ztimer(sema,  (timeout != 0), ZTIMER_USEC, timeout);
#else
    return _sema_wait_xtimer(sema, (timeout != 0), timeout);
#endif
}
#endif
```

Reply:
This was answered in https://github.com/RIOT-OS/RIOT/pull/15782#discussion_r604705742, not a good idea.